### PR TITLE
Document PopStateEvent() and PopStateEvent.state

### PR DESCRIPTION
**Pull request by teoli2003**
### Description

This PR adds the missing documentation for the following:

- `PopStateEvent()`
- `PopStateEvent.state`

### Motivation

openwebdocs/project#153

### Additional details

- I didn't add an example to `PopStateEvent()` as the web developer usually does not call this constructor.
- I removed all uses of `window.history` throughout MDN: modern usage is the plain `history` global variable. (One is left in the `Window` interface where it belongs).
- I added notes indicating that the `History API` can only be used in the window global scope (the main thread) and not in workers or worklets.
- I put `PopStateEvent` inside the _History API_ group.
- There would be many more rabbit holes to follow (the guides could be vastly improved), but that's outside the project's scope.
- The flaws listed are non-written pages unrelated to this work that are in pages I only marginally updated.

### Related pull request

- BCD update: mdn/browser-compat-data#19253

### CODE REVIEW

File: files/en-us/web/api/history_api/example/index.md [hunk 1]
```diff
@@ -1,10 +1,10 @@
 ---
-title: Ajax navigation example
+title: History navigation example
 slug: Web/API/History_API/Example
 page-type: guide
 ---
 
-This is an example of an AJAX website composed only of three pages (_first_page.php_, _second_page.php_ and _third_page.php_). To see how it works, please create the following files (or git clone [https://github.com/giabao/mdn-ajax-nav-example.git](https://github.com/giabao/mdn-ajax-nav-example) ):
+This is an example of anwebsite composed only of three pages (_first_page.php_, _second_page.php_ and _third_page.php_). To see how it works, please create the following files (or git clone [https://github.com/giabao/mdn-ajax-nav-example.git](https://github.com/giabao/mdn-ajax-nav-example) ):
```
Comment (dawei-wang) [inline files/en-us/web/api/history_api/example/index.md:7]:
```suggestion
This is an example of a website composed only of three pages (_first_page.php_, _second_page.php_ and _third_page.php_). To see how it works, please create the following files (or git clone [https://github.com/giabao/mdn-ajax-nav-example.git](https://github.com/giabao/mdn-ajax-nav-example) ):
```

File: files/en-us/web/api/history_api/index.md [hunk 2]
```diff
@@ -53,27 +55,29 @@ Another use for the `go()` method is to refresh the current page by either passi
...
-let numberOfEntries = window.history.length;
+const numberOfEntries = history.length;
 ```
 
 ## Interfaces
 
 - {{domxref("History")}}
   - : Allows manipulation of the browser _session history_ (that is, the pages visited in the tab or frame that the current page is loaded in).
+- {{domxref("PopStateEvent")}}
+  - : The interfave of the {{domxref("Window.popstate_event", "popstate")}} event.
```
Comment (dawei-wang) [inline files/en-us/web/api/history_api/index.md:73]:
```suggestion
  - : The interface of the {{domxref("Window.popstate_event", "popstate")}} event.
```

**Review by dawei-wang — COMMENTED** [review]
A couple of typos

File: files/en-us/web/api/history_api/index.md [hunk 3]
```diff
@@ -7,7 +7,9 @@ browser-compat: api.History
 
 {{DefaultAPISidebar("History API")}}
 
-The DOM {{DOMxRef("Window")}} object provides access to the browser's session history (not to be confused for [WebExtensions history](/en-US/docs/Mozilla/Add-ons/WebExtensions/API/history)) through the {{DOMxRef("Window.history","history")}} object. It exposes useful methods and properties that let you navigate back and forth through the user's history, and manipulate the contents of the history stack.
+The **History API** provides access to the browser's session history (not to be confused for [WebExtensions history](/en-US/docs/Mozilla/Add-ons/WebExtensions/API/history)) through the {{DOMxRef("Window.history","history")}} global object. It exposes useful methods and properties that let you navigate back and forth through the user's history, and manipulate the contents of the history stack.
```
Comment (sideshowbarker) [inline files/en-us/web/api/history_api/index.md:10]:
```suggestion
The **History API** provides access to the browser's session history (not to be confused with [WebExtensions history](/en-US/docs/Mozilla/Add-ons/WebExtensions/API/history)) through the {{DOMxRef("Window.history","history")}} global object. It exposes useful methods and properties that let you navigate back and forth through the user's history, and manipulate the contents of the history stack.
```

**Review by sideshowbarker — APPROVED** [review]

**Pull request merged**